packages/xstate-solid/src/createImmutable.ts

```diff
@@ -0,0 +1,114 @@
+import { createStore, Store } from 'solid-js/store';
+import { deepClone } from './deepClone';
+import { batch } from 'solid-js';
+
+export function isWrappable(obj: any): obj is object {
+  let proto;
+  return (
+    obj != null &&
+    typeof obj === 'object' &&
+    (!(proto = Object.getPrototypeOf(obj)) ||
+      proto === Object.prototype ||
+      Array.isArray(obj))
+  );
+}
+
+const resolvePath = (path: any[], obj = {}) =>
+  path.reduce((prev, curr) => prev?.[curr], obj);
+
+const updateStore = <Path extends unknown[]>(
+  nextStore: Store<any>,
+  prevStore: Store<any>,
+  set: (...args: [...Path, any?, any?]) => void,
```

Comment:
q: why do we have two trailing elements here? could we label all elements here?

Reply:
This was the cleanest way I could make Typescript happy with the spread + tuple where Path is different for each set call. It seemed like every solution had a catch that Typescript didn't like.